new-cdc/eventuate-local-java-cdc-connector-unified/src/main/java/io/eventuate/local/unified/cdc/pipeline/dblog/mysqlbinlog/factory/MySqlBinlogCdcPipelineReaderFactory.java

```diff
@@ -1,41 +1,75 @@
 import io.eventuate.local.java.kafka.consumer.EventuateKafkaConsumerConfigurationProperties;
 import io.eventuate.local.java.kafka.producer.EventuateKafkaProducer;
+import io.eventuate.local.mysql.binlog.DebeziumBinlogOffsetKafkaStore;
+import io.eventuate.local.mysql.binlog.MySqlBinaryLogClient;
 import io.eventuate.local.unified.cdc.pipeline.common.BinlogEntryReaderProvider;
+import io.eventuate.local.unified.cdc.pipeline.dblog.common.factory.CommonDbLogCdcPipelineReaderFactory;
+import io.eventuate.local.unified.cdc.pipeline.dblog.common.factory.OffsetStoreFactory;
 import io.eventuate.local.unified.cdc.pipeline.dblog.mysqlbinlog.properties.MySqlBinlogCdcPipelineReaderProperties;
 import org.apache.curator.framework.CuratorFramework;
 
 import javax.sql.DataSource;
 
-public class MySqlBinlogCdcPipelineReaderFactory extends AbstractMySqlBinlogCdcPipelineReaderFactory {
+public class MySqlBinlogCdcPipelineReaderFactory extends CommonDbLogCdcPipelineReaderFactory<MySqlBinlogCdcPipelineReaderProperties, MySqlBinaryLogClient> {
+  public static final String TYPE = "mysql-binlog";
 
   public MySqlBinlogCdcPipelineReaderFactory(CuratorFramework curatorFramework,
                                              BinlogEntryReaderProvider binlogEntryReaderProvider,
                                              EventuateKafkaConfigurationProperties eventuateKafkaConfigurationProperties,
                                              EventuateKafkaConsumerConfigurationProperties eventuateKafkaConsumerConfigurationProperties,
-                                             EventuateKafkaProducer eventuateKafkaProducer) {
+                                             EventuateKafkaProducer eventuateKafkaProducer,
+                                             OffsetStoreFactory offsetStoreFactory) {
 
     super(curatorFramework,
             binlogEntryReaderProvider,
             eventuateKafkaConfigurationProperties,
             eventuateKafkaConsumerConfigurationProperties,
-            eventuateKafkaProducer);
+            eventuateKafkaProducer,
+            offsetStoreFactory);
   }
 
   @Override
-  protected OffsetStore createOffsetStore(MySqlBinlogCdcPipelineReaderProperties properties,
-                                          DataSource dataSource,
-                                          EventuateSchema eventuateSchema,
-                                          String clientName) {
+  public boolean supports(String type) {
+    return TYPE.equals(type);
+  }
 
-    return new DatabaseOffsetKafkaStore(properties.getDbHistoryTopicName(),
-            clientName,
-            eventuateKafkaProducer,
+  @Override
+  public Class<MySqlBinlogCdcPipelineReaderProperties> propertyClass() {
+    return MySqlBinlogCdcPipelineReaderProperties.class;
+  }
+
+  @Override
+  public MySqlBinaryLogClient create(MySqlBinlogCdcPipelineReaderProperties mySqlBinlogCdcPipelineReaderProperties) {
```

Comment:
Please rename the parameter name from mySqlBinlogCdcPipelineReaderProperties to => readerProperties